src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -301,5 +310,47 @@ private static bool TypeMatches(JsonSchemaType schemaType, JTokenType tokenType)
                     return true;
             }
         }
+
+        private static bool JTokenDeepEquals(JToken a, JToken b)
+        {
+            switch (a)
+            {
+                case JValue valueA when b is JValue valueB:
+                    return Equals(valueA.Value, valueB.Value);
+
+                case JArray arrayA when b is JArray arrayB:
+                    if (arrayA.Count != arrayB.Count)
+                    {
+                        return false;
+                    }
+
+                    for (var i = 0; i < arrayA.Count; i++)
+                    {
+                        if (!JTokenDeepEquals(arrayA[i], arrayB[i]))
+                        {
+                            return false;
+                        }
+                    }
+                    return true;
+
+                case JObject mapA when b is JObject mapB:
+                    if (mapA.Count != mapB.Count)
+                    {
+                        return false;
+                    }
+
+                    foreach (var (key, valueA) in mapA)
+                    {
+                        if (!mapB.TryGetValue(key, out var valueB) || !JTokenDeepEquals(valueA, valueB))
```

Comment:
InlineData("{'const': {}}", "{‘a’ : 2}") What is the behavior of this situation? The current logic seems to be no error.

Reply:
It will fail, there is a check for count in line 337, the JSON schema test suite has this case.